t/25-spvm.t

```diff
@@ -36,4 +37,22 @@ subtest 'SSH credentials in spvm' => sub {
     is($spvm->run_cmd('true', $expected_credentials->{hostname}, $expected_credentials->{password}), 0, "Test specific credentials");
 };
 
+subtest 'PowerVM power actions' => sub {
+    my $mock_spvm = Test::MockModule->new('backend::spvm');
+    $mock_spvm->mock('run_cmd', sub {
+            my ($self, $cmd) = @_;
+            return $cmd;
+    });
+
+    my $spvm    = backend::spvm->new();
+    my $lpar_id = 3;
+    set_var(NOVALINK_LPAR_ID => $lpar_id);
+
+    my %actions = (on => "pvmctl lpar power-on -i id=${lpar_id} --bootmode norm",
+        off   => "pvmctl lpar power-off -i id=${lpar_id} --hard",
+        reset => "pvmctl lpar restart -i id=${lpar_id}");
+    while (my ($action, $result) = each %actions) {
```

Comment:
Don't use `each`. It is said to be slow. I'm also wondering whether it is required to test all cases here. This test just feels like repeating the declaration of the data structure. In an integration/fullstack test it would make sense to run all cases to see whether the commands actually have the effect we think they have but here I see no additional benefit.

However, it would make sense to test the `die` case as it is special. (I guess there's a test helper to test for exceptions.)

Reply:
>In an integration/fullstack test it would make sense to run all cases to see whether the commands actually have the effect we think they have but here I see no additional benefit.

Not sure If I understand. Do you mean that I should implement behavior of pvmctl command in test? Or should I test only one action and die result?

> However, it would make sense to test the `die` case as it is special. (I guess there's a test helper to test for exceptions.)
OK, clear.

Reply:
@Martchus replaced `each` by `foreach`, added test for die exception

Reply:
> Not sure If I understand. Do you mean that I should implement behavior of pvmctl command in test? Or should I test only one action and die result?

The 2nd option.

Reply:
It should be done.